samcli/lib/providers/sam_function_provider.py

```diff
@@ -79,6 +79,26 @@ def stacks(self) -> List[Stack]:
         """
         return self._stacks
 
+    def update(
```

Comment:
Would it be possible to call this `update()` function in `__init__()` to remove the repeat code? Otherwise this LGTM

Reply:
Not sure if I'm following. The provider doesn't get re-instantiated after the first time we instantiate it in the `BuildContext`, so how would it update?

Reply:
Its more so that the new update function and __init__ both extract the functions from the stack, so I was wondering if we could just call `self.update()` inside of the constructor instead of having `self.functions = SamFunctionProvider._extract_functions(...)` in two places.

Reply:
The only thing with this is that instance variables should be instantiated in the constructor otherwise our linter will complain.